auth_from_http_remote_user_role/controllers/main.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import http, registry
+from odoo.http import request
+
+from odoo.addons.auth_from_http_remote_user.controllers import main
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class Home(main.Home):
+
+    _REMOTE_USER_ROLE = 'HTTP_USER_ROLES'
+    _REMOTE_USER_ROLE_SEPARATOR = ','
+
+    def _get_roles(self):
+        """Return user roles from HTTP header.
+
+        Return in a list the user role codes found in the HTTP
+        header using the field and separator predefined.
+        """
+        roles = []
+        headers = http.request.httprequest.headers.environ
+        roles_in_header = headers.get(self._REMOTE_USER_ROLE, None)
+        if roles_in_header:
+            roles = roles_in_header.split(self._REMOTE_USER_ROLE_SEPARATOR)
+        return roles
+
+    def _authenticate_user(self, user, cr):
```

Comment:
What is the reason to use direct SQL rather than the ORM? if there is a reason, could you document it as comments? Besides, cache must be invalidated after the executes.